### PR TITLE
refactor(v3): Remove all references to temporary downloads folder

### PR DESCRIPTION
**Pull request by traeok**
## Proposed changes

Removes the following settings and any code related to the temp folder (creation, cleanup, etc.):
image

## Release Notes

Milestone: vNext

Changelog:
- **Breaking:** Zowe Explorer no longer uses a temporary directory for storing Data Sets and USS files. All settings related to the temporary downloads folder have been removed. In order to access resources stored by Zowe Explorer v3, refer to the [FileSystemProvider documentation](https://github.com/zowe/zowe-explorer-vscode/wiki/FileSystemProvider) for information on how to build and access resource URIs. Extenders can detect changes to resources using the `onResourceChanged` function in the `ZoweExplorerApiRegister` class.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (non-breaking change which adds or improves functionality)
- [x] Breaking change (a change that would cause existing functionality to not work as expected)
- [x] Documentation (Markdown, README updates)
- [ ] Other (please specify above in "Proposed changes" section)

## Checklist

**General**

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/zowe-explorer-vscode/wiki/Best-Practices:-Contributor-Guidance) wiki
- [ ] A GIF or screenshot is included in the PR for visual changes
- [x] The pre-publish command has been executed:
  - **v2 and below:** `yarn workspace vscode-extension-for-zowe vscode:prepublish`
  - **v3:** `pnpm --filter vscode-extension-for-zowe vscode:prepublish`

**Code coverage**

- [x] There is coverage for the code that I have added
- [x] I have added new test cases and they are passing
- [x] I have manually tested the changes

**Deployment**

- [x] I have added developer documentation (if applicable)
- [ ] Documentation should be added to Zowe Docs
  - If you're an outside contributor, please post in the [#zowe-doc Slack channel](https://openmainframeproject.slack.com/archives/CC961JYMQ) to coordinate documentation.
  - Otherwise, please check with the rest of the squad about any needed documentation before merging.
- [ ] These changes may need ported to the appropriate branches (list here):

### CODE REVIEW

**Comment by traeok** [comment]
SonarCloud has spotted existing code duplication from previous contributions. Existing code duplication can be addressed in a separate PR.

File: packages/zowe-explorer/CHANGELOG.md [hunk 1]
```diff
@@ -36,6 +36,7 @@ All notable changes to the "vscode-extension-for-zowe" extension will be documen
   - `zowe.ds.ssoLogin`, `zowe.uss.ssoLogin`, `zowe.jobs.ssoLogin` - use `zowe.ssoLogin` instead
   - `zowe.ds.ssoLogout`, `zowe.uss.ssoLogout`, `zowe.jobs.ssoLogout` - use `zowe.ssoLogout` instead
 - Added PEM certificate support as an authentication method for logging into the API ML. [#2621](https://github.com/zowe/zowe-explorer-vscode/issues/2621)
+- **Breaking:** Zowe Explorer no longer uses a temporary directory for storing Data Sets and USS files. All settings related to the temporary downloads folder have been removed. In order to access resources stored by Zowe Explorer v3, refer to the [FileSystemProvider documentation](https://github.com/zowe/zowe-explorer-vscode/wiki/FileSystemProvider) for information on how to build and access resource URIs. Extenders can detect changes to resources using the `onResourceChanged` function in the `ZoweExplorerApiRegister` class.
```
Comment (anaxceron) [inline packages/zowe-explorer/CHANGELOG.md:39]: Is it possible to include a pr/issue link with this?

Comment (traeok) [reply packages/zowe-explorer/CHANGELOG.md]: Yep, sorry forgot to add - addressed in commit ad328d2 👍

**Review by anaxceron — APPROVED** [review]
Looks good, thanks @traeok!

**Review by zFernand0 — DISMISSED** [review]
Hey @traeok,
Thanks for adding new tests for sumbitJcl. 🥳 
Curious if we should check this box. 😋 
> - [ ] I have added new test cases and they are passing

LGTM! 😋

**Review by t1m0thyj — DISMISSED** [review]
LGTM, thanks @traeok!

**Review by zFernand0 — APPROVED** [review]
Changes still LGTM! 😋

**Review by t1m0thyj — DISMISSED** [review]

**Review by zFernand0 — APPROVED** [review]

**Review by t1m0thyj — APPROVED** [review]

**Pull request merged**